pycqed/analysis_v2/timedomain_analysis.py

```diff
@@ -759,6 +769,29 @@ def rotate_data_TwoD(qb_name, meas_results_per_qb, channel_map,
                                 cal_one_points=cal_one_points)
         return rotated_data_dict
 
+    def get_xaxis_label_unit(self, qb_name):
+        # get xunit and label: temporarily with try catch for old
+        # sequences which do not have unit and label in meta data
+        hard_sweep_params = self.get_param_value('hard_sweep_params')
+        sweep_name = self.get_param_value('sweep_name')
+        sweep_unit = self.get_param_value('sweep_unit')
+        if self.sp is not None:
+            _, xunit, xlabel = self.sp.get_sweep_params_description(
+                param_names=self.mospm[qb_name], dimension=1)[0]
+        elif hard_sweep_params is not None:
+            xlabel = list(hard_sweep_params)[0]
+            xunit = list(hard_sweep_params.values())[0][
+                'unit']
+        elif (sweep_name is not None) and (sweep_unit is not None):
+            xlabel = sweep_name
+            xunit = sweep_unit
+        else:
+            xlabel = self.raw_data_dict['sweep_parameter_names']
```

Comment:
No because these entries will always exist because they are created by the MC and because the extraction of these parameters is hardcoded in the BaseDataAnalysis .